packages/next/src/server/base-server.ts

```diff
@@ -3421,6 +3421,12 @@ export default abstract class Server<
       // don't bother to respond with 404, because these requests are only
       // issued as part of a prefetch.
       res.statusCode = 204
+
+      res.setHeader(
+        'Cache-Control',
+        'private, no-cache, no-store, max-age=0, must-revalidate'
+      )
```

Comment:
Do we also need to clear the `cacheEntry?.cacheControl` field below?

Reply:
Nice catch, didn't see that